lib/galaxy/model/tags.py

```diff
@@ -151,6 +154,20 @@ def delete_item_tags(self, user, item):
         # Delete tags from item.
         del item.tags[:]
 
+    def _ensure_user_owns_item(self, user, item):
+        """Raises exception if user does not own item.
+        Notice that even admin users cannot directly modify tags on items they do not own.
+        To modify tags on items they don't own, admin users must impersonate the item's owner.
+        """
+        if getattr(item, "id", None) is None:
+            # Item is not persisted, likely it is being copied from an existing, so no need
+            # to check ownership at this point.
+            return
+        history = getattr(item, "history", None)
+        is_owner = getattr(item, "user", None) == user or getattr(history, "user", None) == user
+        if not is_owner:
```

Comment:
I'll link an open comment from Marius here as the comment is on a rebased commit that Github doesn't like to display on the PR.

https://github.com/galaxyproject/galaxy/commit/280004bb3b4fae08b0d5f9227c2dd2b83d2f4943#r120498112